src/django/api/serializers.py

```diff
@@ -493,13 +495,73 @@ def format_source(source):
         return distinct_sources
 
 
+def parse_raw_data(data):
+    try:
+        # try to parse as json
+        return json.loads(data)
+    except json.decoder.JSONDecodeError:
+        try:
+            # try to parse as stringified object
+            return json.loads(data.replace("'", '"'))
+        except json.decoder.JSONDecodeError:
+            return {}
+
+
+def get_csv_values(csv_data):
+    values = []
+    csvreader = csv.reader(csv_data.split('\n'), delimiter=',')
+    for row in csvreader:
+        values = values + row
+    return values
+
+
+def get_single_contributor_field_values(item, fields):
+    data = parse_raw_data(item.raw_data)
+    for f in fields:
+        value = data.get(f['column_name'], None)
+        if value is not None:
+            f['values'].append(value)
+    return fields
+
+
+def get_list_contributor_field_values(item, fields):
+    data_values = get_csv_values(item.raw_data)
+    list_fields = get_csv_values(item.source.facility_list.header)
+    for f in fields:
+        if f['column_name'] in list_fields:
+            index = list_fields.index(f['column_name'])
+            f['values'].append(data_values[index])
+    return fields
+
+
+def assign_contributor_field_values(list_items, fields):
+    contributor_fields = [{
+        'label': f.display_name, 'values': [], 'column_name': f.column_name
+    } for f in fields]
+
+    for item in list_items:
+        if item.source.source_type == 'SINGLE':
+            contributor_fields = get_single_contributor_field_values(
+                                    item, contributor_fields
+                                )
+        else:
+            contributor_fields = get_list_contributor_field_values(
+                                    item, contributor_fields
+                                 )
+
+    return [{
+                'values': f['values'], 'label': f['label']
+            } for f in contributor_fields]
```

Comment:
Although the same facility could be submitted multiple times and therefore have multiple line items matched to it, I think that we should assume that the most recent, active submission is the one from which we should pull the nonstandard field values. I mentioned this at the point in the code where the QuerySet is built and, as a result, we should be able to take a single list item as a parameter to this function and use a single `'value': None` to initialize the return rather than `'values': []'`.